src/nwb_conversion_tools/tools/spikeinterface/spikeinterface.py

```diff
@@ -1118,51 +1117,108 @@ def add_units_table(
 
     property_descriptions = dict(default_descriptions, **property_descriptions)
 
-    all_properties = checked_sorting.get_property_keys()
-    write_properties = set(all_properties) - set(skip_properties)
-    for property_name in write_properties:
-        if property_name not in property_descriptions:
-            warnings.warn(
-                f"Description for property {property_name} not found in property_descriptions. "
-                "Setting description to 'no description'"
-            )
-    aggregated_unit_properties = defaultdict()
-    for property_name in write_properties:
-        unit_col_args = dict(
-            name=property_name, description=property_descriptions.get(property_name, "No description.")
-        )
-        if property_name in ["max_channel", "max_electrode"] and nwbfile.electrodes is not None:
-            unit_col_args.update(table=nwbfile.electrodes)
+    data_to_add = defaultdict(dict)
+    sorting_properties = checked_sorting.get_property_keys()
+    excluded_properties = list(skip_properties) + ["contact_vector"]
+    properties_to_extract = [property for property in sorting_properties if property not in excluded_properties]
 
-        units_table.add_column(**unit_col_args)
-        aggregated_unit_properties[property_name] = checked_sorting.get_property(key=property_name)
+    # Extract properties
+    for property in properties_to_extract:
+        data = checked_sorting.get_property(property)
+        index = isinstance(data[0], (list, np.ndarray, tuple))
+        description = property_descriptions.get(property, "No description.")
+        data_to_add[property].update(description=description, data=data, index=index)
+        if property in ["max_channel", "max_electrode"] and nwbfile.electrodes is not None:
+            data_to_add[property].update(table=nwbfile.electrodes)
 
     # Unit name logic
     units_ids = checked_sorting.get_unit_ids()
-    if "unit_name" in aggregated_unit_properties:
-        unit_name_array = aggregated_unit_properties["unit_name"]
+    if "unit_name" in data_to_add:
+        unit_name_array = data_to_add["unit_name"]["data"]
     else:
         unit_name_array = units_ids.astype("str", copy=False)
-        unit_col_args = dict(
-            name="unit_name",
-            description="unique reference for each unit",
-        )
-        units_table.add_column(**unit_col_args)
-        aggregated_unit_properties["unit_name"] = unit_name_array
+        data_to_add["unit_name"].update(description="unique reference for each unit", data=unit_name_array, index=False)
```

Comment:
Let's try our best to keep `descriptions` to be proper sentences (start with capital, end with punctuation). It's too much to require it or even have it be a best practice, but there's no excuse not to do it for anything we set automatically.

Reply:
Also, is the `index=False` explicitly necessary? Shouldn't it be `False` by default? Ditto for below with the `id` key.